src/sedtrails/logger/logger.py

```diff
@@ -8,35 +8,125 @@
-    
-    :param state: Dictionary containing relevant simulation data.
-    :param level: Logging level (e.g., logging.INFO, logging.DEBUG)
+    Logs the current state of the simulation with human-readable sentences.
     """
-    message = ", ".join(f"{key}={value}" for key, value in state.items())
-    logger.log(level, f"Simulation State: {message}")
+    logger = _logger_manager.setup_logger()
+    
+    status = state.get('status', state.get('state', 'unknown'))
+    
+    # Create full sentence messages
+    if status == 'simulation_started':
+        command = state.get('command', 'unknown')
+        config = state.get('config_file', 'unknown')
+        python_ver = state.get('python_version', 'unknown')
+        message = f"Starting SedTRAILS simulation with command: '{command}' using Python {python_ver} and config: {config}"
+        
+    elif status == 'config_loading':
+        config_path = state.get('config_file_path', 'unknown')
+        message = f"Loading configuration from: {config_path}"
+        
+    elif status == 'simulation_parameters':
+        duration = state.get('duration', 'unknown')
+        timestep = state.get('timestep', 'unknown')
+        start_time = state.get('start_time', 'unknown')
+        strategy = state.get('seeding_strategy', 'unknown')
+        output_dir = state.get('output_dir', 'unknown')
+        message = f"Simulation parameters: duration={duration}, timestep={timestep}, start_time={start_time}, seeding={strategy}, output={output_dir}"
+        
+    elif status == 'particles_created' or status == 'particles_initialized':
+        count = state.get('count', 1)
+        position = state.get('start_position', state.get('position', 'unknown'))
+        message = f"Created {count} particle(s) starting at position {position}"
+        
+    elif status == 'compilation_complete':
+        time_sec = state.get('time_sec', 'unknown')
+        message = f"Particle calculator compiled successfully in {time_sec} seconds"
+        
+    elif status == 'warmup_complete':
+        time_sec = state.get('time_sec', 'unknown')
+        message = f"JIT compiler warmed up in {time_sec} seconds"
+        
+    elif status == 'simulation_progress':
+        progress = state.get('progress_pct', 0)
+        step = state.get('step', 0)
+        position = state.get('position', 'unknown')
+        message = f"Simulation {progress}% complete (step {step}) - particle at {position}"
+        
+    elif status == 'simulation_complete' or status == 'simulation_completed':
+        steps = state.get('total_steps', 'unknown')
+        time_sec = state.get('total_time_sec', 'unknown')
+        final_pos = state.get('final_position', 'unknown')
+        message = f"Simulation completed! {steps} steps in {time_sec}s, final position: {final_pos}"
+        
+    elif status == 'visualization_saved':
+        filename = state.get('file', 'trajectory plot')
+        message = f"Trajectory visualization saved as {filename}"
```

Comment:
These messages would need to change. You can leave it as they are here. However, keep in mind that for example, messages for the visualisation aren't relevant because in our architecture, visualisations is handle by a separated component.
in other words, I expect to remove the code that create plots from the Simualtion class.